File: frontend/src/app/model/response/AgendamentoResponse.ts
export interface AgendamentoResponse {
  id: string;
  estabelecimento: {
    id: string;
    nome: string;
    urlFotoPerfil: string;
    email: string;
    tipo: number;
    cnpj: string;
    descricao: string;
  };
  cliente: {
    id: string; // UUID
    nome: string;
    urlFotoPerfil: string;
    email: string;
    tipo: number;
    cpf: string;
  };
  servico: {
    id: string;
    estabelecimentoId: string;
    nome: string;
    duracao: string;
  };
  data: string;
  status: string;
}

File: domain/src/main/java/com/agendify/domain/records/AgendamentoResponse.java
package com.agendify.domain.records;

import com.agendify.domain.entities.Status;
import com.agendify.domain.records.ClienteResponse;
import com.agendify.domain.records.EstabelecimentoResponse;
import com.agendify.domain.records.ServicoResponse;

import java.util.Date;
import java.util.UUID;

public record AgendamentoResponse(UUID id,
                                  EstabelecimentoResponse estabelecimento,
                                  ClienteResponse cliente,
                                  ServicoResponse servico,
                                  Date data,
                                  Status status) {
}
